modules/apps/object/object-test/src/testIntegration/java/com/liferay/object/service/test/ObjectFieldLocalServiceTest.java

```diff
@@ -955,6 +956,71 @@ public void testObjectFieldSettings() throws Exception {
 			objectDefinition.getObjectDefinitionId());
 	}
 
+	@Test
+	public void testRelationshipWithMandatoryObjectField() throws Exception {
+		ObjectDefinition objectDefinition1 =
+			ObjectDefinitionTestUtil.addObjectDefinition(
+				_objectDefinitionLocalService,
+				Arrays.asList(
+					ObjectFieldUtil.createObjectField(
+						ObjectFieldConstants.BUSINESS_TYPE_TEXT,
+						ObjectFieldConstants.DB_TYPE_STRING,
+						RandomTestUtil.randomString(), StringUtil.randomId())));
+
+		objectDefinition1 =
+			_objectDefinitionLocalService.publishCustomObjectDefinition(
```

Comment:
Are you publishing these definitions for any specific reason?

Reply:
I am publishing these definitions because we can change the mandatory relationship after publish _objectDefinition_. So, we need test that change after publishing. It make sense?

Reply:
Yeah it makes sense. Do you need to publish both objects or just one?
Also, could you include a case where you verify if you can update the required before publishing as well? Thanks!

Reply:
Hi @rodrigopaulino, changes made! Thanks so much